dali/kernels/imgproc/convolution/convolution_gpu.h

```diff
@@ -29,6 +30,29 @@
 namespace dali {
 namespace kernels {
 
+struct ConvEpilogue {
+  ConvEpilogue(span<const float> scales, float beta = 0.f)  // NOLINT
+      : scales_{scales}, scale_{}, beta_{beta} {}
+  ConvEpilogue(float scale, float beta = 0.f) : scales_{}, scale_{scale}, beta_{beta} {}  // NOLINT
+
+  inline float num_samples() const {
+    return scales_.empty() ? 1 : scales_.size();
```

Comment:
Why so? I think some special value - or just 0 - would be preferable. Note that you can query for an arbitrary `alpha`  index when scales are empty, but the return value of this function suggests otherwise.